src/analysis/object.rs

```diff
@@ -66,23 +66,23 @@ pub fn class(env: &Env, obj: &GObject, deps: &[library::TypeId]) -> Option<Info>
 
     let supertypes = supertypes::analyze(env, class_tid, &mut imports);
 
-    let mut has_children = obj.force_trait;
-
-    for child_tid in env.class_hierarchy.subtypes(class_tid) {
-        let child_name = child_tid.full_name(&env.library);
-        let status = env.config.objects.get(&child_name)
-            .map(|o| o.status)
-            .unwrap_or_default();
-        if status.normal() {
-            has_children = true;
-            break;
+    let mut generate_trait = obj.generate_trait;
+
+    // Sanity check the user's configuration. It's unlikely that not generating
+    // a trait is wanted if there are subtypes in this very crate
+    if !generate_trait {
+        for child_tid in env.class_hierarchy.subtypes(class_tid) {
+            let child_name = child_tid.full_name(&env.library);
+            let status = env.config.objects.get(&child_name)
+                .map(|o| o.status)
+                .unwrap_or_default();
+            if status.normal() {
+                error!("Not generating trait for {} although subtypes exist", full_name);
+                break;
+            }
```

Comment:
IMHO this better separate function like `has_subtypes` here.
But I don't insist this